src/main/kotlin/org/jitsi/nlj/transform/node/incoming/VideoParser.kt

```diff
@@ -61,51 +67,78 @@ class VideoParser(
                     val vp8Packet = packetInfo.packet.toOtherType(::Vp8Packet)
                     packetInfo.packet = vp8Packet
                     packetInfo.resetPayloadVerification()
+
+                    if (videoCodecParser !is Vp8Parser) {
+                        resetSources()
+                        videoCodecParser = Vp8Parser(sources, logger)
+                    }
+                }
+                is Vp9PayloadType -> {
+                    val vp9Packet = packetInfo.packet.toOtherType(::Vp9Packet)
+                    packetInfo.packet = vp9Packet
+                    packetInfo.resetPayloadVerification()
+
+                    if (videoCodecParser !is Vp9Parser) {
+                        resetSources()
+                        videoCodecParser = Vp9Parser(sources, logger)
+                    }
+                }
+                else -> {
+                    videoCodecParser = null
                 }
             }
+            videoCodecParser?.parse(packetInfo)
+            packet = packetInfo.packetAs()
```

Comment:
Rather than reassign `packet` I think it'd be cleaner to have the `when` return the parsed packet and assign another `val` there.  What type is the bare `packetAs` even giving you there?